examples/ShowLoader.cpp

```diff
@@ -141,6 +141,24 @@ ShowLoader::State ShowLoader::NextFrame(unsigned int *universe,
 }
 
 
+/**
+ * Read the next show file entry
+ * @param entry a ShowEntry to fill with data
+ */
+ShowLoader::State ShowLoader::NextEntry(ShowEntry *entry) {
+  State state = NextFrame(&entry->universe, &entry->buffer);
+  if (state != State::OK) {
+      return state;
+  }
+  state = NextTimeout(&entry->next_wait);
+  if (state != State::OK) {
+      return state;
+  }
+
+  return State::OK;
```

Comment:
Any reason to not just return state here? Which also makes the if above redundant for now, aside from my other comment.